.travis.yml

```diff
@@ -0,0 +1,33 @@
+services:
+- docker
+
+before_install:
+  - echo "$DOCKER_HUB_PASSWORD" | docker login --username $DOCKER_HUB_USERNAME --password-stdin
+  - curl https://cli-assets.heroku.com/install.sh | sh  # install heroku
+  - docker login --username=$HEROKU_LOGIN --password=$HEROKU_API_KEY registry.heroku.com 
+
+before_script:
+  - docker build --target test --tag test .
+  - docker build --tag $DOCKER_HUB_USERNAME/todo-app:latest --target production .
+  - docker tag $DOCKER_HUB_USERNAME/todo-app:latest registry.heroku.com/$HEROKU_APP/web
```

Comment:
If these steps are just for the Heroku deployment, they could be moved to a before_deploy stage.